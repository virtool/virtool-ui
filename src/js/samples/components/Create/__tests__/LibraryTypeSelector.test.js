import { LibraryTypeSelector } from "../LibraryTypeSelector";

describe("<LibraryTypeSelector>", () => {
    let props;
    beforeEach(() => {
        props = {
            onSelect: jest.fn(),
            libraryType: "normal"
        };
    });

    it("should have Normal selected [libraryType='normal']", () => {
        const wrapper = shallow(<LibraryTypeSelector {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should have sRNA selected when [libraryType='srna']", () => {
        props.libraryType = "srna";
        const wrapper = shallow(<LibraryTypeSelector {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should have Amplicon selected when [libraryType='amplicon']", () => {
        props.libraryType = "amplicon";
        const wrapper = shallow(<LibraryTypeSelector {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should call onSelect with libraryType normal when Normal type is clicked", () => {
        const { getByText } = renderWithProviders(<LibraryTypeSelector {...props} />);
        fireEvent.click(getByText("Normal"));
        expect(props.onSelect).toHaveBeenCalledWith("normal");
    });
    it("should call onSelect with libraryType srna when srna type is clicked", () => {
        const { getByText } = renderWithProviders(<LibraryTypeSelector {...props} />);
        fireEvent.click(getByText("sRNA"));
        expect(props.onSelect).toHaveBeenCalledWith("srna");
    });
    it("should call onSelect with libraryType amplicon when amplicon type is clicked", () => {
        const { getByText } = renderWithProviders(<LibraryTypeSelector {...props} />);
        fireEvent.click(getByText("Amplicon"));
        expect(props.onSelect).toHaveBeenCalledWith("amplicon");
    });
});
