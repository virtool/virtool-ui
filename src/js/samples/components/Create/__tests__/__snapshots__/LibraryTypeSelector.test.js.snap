// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LibraryTypeSelector> should have Amplicon selected when [libraryType='amplicon'] 1`] = `
<Fragment>
  <label>
    Library Type
  </label>
  <LibraryTypeSelector__LibraryTypeSelectBoxContainer>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Normal
      </div>
      <span>
        Search against whole genome references using normal reads.
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        sRNA
      </div>
      <span>
        Search against whole genome references using sRNA reads
      </span>
    </SelectBox>
    <SelectBox
      active={true}
      onClick={[Function]}
    >
      <div>
        Amplicon
      </div>
      <span>
        Search against barcode references using amplicon reads.
      </span>
    </SelectBox>
  </LibraryTypeSelector__LibraryTypeSelectBoxContainer>
</Fragment>
`;

exports[`<LibraryTypeSelector> should have Normal selected [libraryType='normal'] 1`] = `
<Fragment>
  <label>
    Library Type
  </label>
  <LibraryTypeSelector__LibraryTypeSelectBoxContainer>
    <SelectBox
      active={true}
      onClick={[Function]}
    >
      <div>
        Normal
      </div>
      <span>
        Search against whole genome references using normal reads.
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        sRNA
      </div>
      <span>
        Search against whole genome references using sRNA reads
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Amplicon
      </div>
      <span>
        Search against barcode references using amplicon reads.
      </span>
    </SelectBox>
  </LibraryTypeSelector__LibraryTypeSelectBoxContainer>
</Fragment>
`;

exports[`<LibraryTypeSelector> should have sRNA selected when [libraryType='srna'] 1`] = `
<Fragment>
  <label>
    Library Type
  </label>
  <LibraryTypeSelector__LibraryTypeSelectBoxContainer>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Normal
      </div>
      <span>
        Search against whole genome references using normal reads.
      </span>
    </SelectBox>
    <SelectBox
      active={true}
      onClick={[Function]}
    >
      <div>
        sRNA
      </div>
      <span>
        Search against whole genome references using sRNA reads
      </span>
    </SelectBox>
    <SelectBox
      active={false}
      onClick={[Function]}
    >
      <div>
        Amplicon
      </div>
      <span>
        Search against barcode references using amplicon reads.
      </span>
    </SelectBox>
  </LibraryTypeSelector__LibraryTypeSelectBoxContainer>
</Fragment>
`;
